pkg/kubectl/cmd/util/cached_discovery_test.go

```diff
@@ -85,22 +92,29 @@ func TestNewCachedDiscoveryClient_TTL(t *testing.T) {
 	assert.NoError(err)
 	defer os.RemoveAll(d)
 
-	c := fakeDiscoveryClient{}
-	cdc := NewCachedDiscoveryClient(&c, d, 1*time.Nanosecond)
+	c := NewFakeDiscoveryClient()
+	cdc := NewCachedDiscoveryClient(c, d, 1*time.Nanosecond)
 	cdc.ServerGroups()
-	assert.Equal(c.groupCalls, 1)
+	assert.Equal(c.GroupCalls, 1)
 
 	time.Sleep(1 * time.Second)
 
 	cdc.ServerGroups()
-	assert.Equal(c.groupCalls, 2)
+	assert.Equal(c.GroupCalls, 2)
+}
+
+func NewFakeDiscoveryClient() *fakeDiscoveryClient {
+	return &fakeDiscoveryClient{ServerResourcesForGroupVersionHandler: nil, ServerResourcesHandler: nil}
 }
 
 type fakeDiscoveryClient struct {
-	groupCalls    int
-	resourceCalls int
-	versionCalls  int
-	swaggerCalls  int
+	GroupCalls    int
```

Comment:
good catch - no need to export - my original idea was to move the code outside the package.